govc/sso/service/ls.go

```diff
@@ -0,0 +1,137 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package service
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"text/tabwriter"
+
+	"github.com/vmware/govmomi/govc/cli"
+	"github.com/vmware/govmomi/govc/flags"
+	"github.com/vmware/govmomi/lookup"
+	"github.com/vmware/govmomi/lookup/types"
+)
+
+type ls struct {
+	*flags.ClientFlag
+	*flags.OutputFlag
+
+	long bool
+
+	types.LookupServiceRegistrationFilter
+}
+
+func init() {
+	cli.Register("sso.service.ls", &ls{})
+}
+
+func (cmd *ls) Register(ctx context.Context, f *flag.FlagSet) {
+	cmd.ClientFlag, ctx = flags.NewClientFlag(ctx)
+	cmd.ClientFlag.Register(ctx, f)
+
+	cmd.OutputFlag, ctx = flags.NewOutputFlag(ctx)
+	cmd.OutputFlag.Register(ctx, f)
+
+	f.BoolVar(&cmd.long, "l", false, "Long listing format")
+
+	cmd.LookupServiceRegistrationFilter.EndpointType = new(types.LookupServiceRegistrationEndpointType)
+	cmd.LookupServiceRegistrationFilter.ServiceType = new(types.LookupServiceRegistrationServiceType)
+	f.StringVar(&cmd.SiteId, "s", "", "Site ID")
+	f.StringVar(&cmd.NodeId, "n", "", "Node ID")
+	f.StringVar(&cmd.ServiceType.Type, "t", "", "Service type")
+	f.StringVar(&cmd.ServiceType.Product, "p", "", "Service product")
+	f.StringVar(&cmd.EndpointType.Type, "T", "", "Endpoint type")
+	f.StringVar(&cmd.EndpointType.Protocol, "P", "", "Endpoint protocol")
+}
+
+func (cmd *ls) Description() string {
+	return `List platform services.
+
+Examples:
+  govc sso.service.ls
+  govc sso.service.ls -t vcenterserver -P vmomi
+  govc sso.service.ls -t sso:sts
+  govc sso.service.ls -t sso:sts -json | jq -r .[].ServiceEndpoints[].Url`
+}
+
+func (cmd *ls) Process(ctx context.Context) error {
+	if err := cmd.ClientFlag.Process(ctx); err != nil {
+		return err
+	}
+	return cmd.OutputFlag.Process(ctx)
+}
+
+type infoResult []types.LookupServiceRegistrationInfo
+
+func (r infoResult) Dump() interface{} {
+	return []types.LookupServiceRegistrationInfo(r)
+}
+
+func (r infoResult) Write(w io.Writer) error {
+	tw := tabwriter.NewWriter(w, 2, 0, 2, ' ', 0)
+
+	for _, info := range r {
+		fmt.Fprintf(tw, "%s\t%s\t%s\n", info.ServiceType.Product, info.ServiceType.Type, info.ServiceId)
+	}
+
+	return tw.Flush()
+}
+
+type infoResultLong []types.LookupServiceRegistrationInfo
+
+func (r infoResultLong) Dump() interface{} {
+	return []types.LookupServiceRegistrationInfo(r)
+}
+
+func (r infoResultLong) Write(w io.Writer) error {
+	tw := tabwriter.NewWriter(w, 2, 0, 2, ' ', 0)
+
+	for _, info := range r {
+		for _, s := range info.ServiceEndpoints {
+			fmt.Fprintf(tw, "%s\t%s\t%s\t%s\t%s\t%s\n",
+				info.ServiceType.Product, info.ServiceType.Type, info.ServiceId,
+				s.EndpointType.Protocol, s.EndpointType.Type, s.Url)
```

Comment:
Hi @dougm,

I realized it would moan about the generated code, but as I wasn't sure which was entirely generated and which wasn't (outside of the types file), I mentioned it here just once in case.